tools/c7n_oci/c7n_oci/policy.py

```diff
@@ -0,0 +1,168 @@
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import os
+import time
+
+from c7n_oci import mu
+
+from c7n import utils
+from c7n.exceptions import PolicyValidationError
+from c7n.policy import ServerlessExecutionMode, execution
+from c7n.utils import type_schema
+
+
+class FunctionMode(ServerlessExecutionMode):
+    schema = {
+        'type': 'object',
+        'additionalProperties': False,
+        'properties': {
+            'execution-options': {'type': 'object'},
+            'subnets': {'type': 'array'},
+            'function-prefix': {'type': 'string'},
+            'timeout': {'type': 'number'},
+            'memory': {'type': 'number'},
+            'freeform_tags': {'type': 'object'},
+        },
+        'required': ['subnets'],
+    }
+
+    def __init__(self, policy):
+        self.policy = policy
+        self.log = logging.getLogger('custodian.oci.functionexec')
+
+    def run(self):
+        raise NotImplementedError("subclass responsibility")
+
+    def provision(self):
+        with self.policy.ctx:
+            self.policy.log.info("Provisioning policy function %s", self.policy.name)
+            manager = mu.FunctionManager(self.policy.session_factory)
+            return manager.publish(self.get_function())
+
+    def deprovision(self):
+        self.log.info("Removing policy function %s", self.policy.name)
+        manager = mu.FunctionManager(self.policy.session_factory)
+        return manager.remove(self.get_function())
+
+    def validate(self):
```

Comment:
validate should also check for presence of docker module, so it doesn't fail at runtime/policy execution.

Reply:
Added code for checking the docker presence.